detect-nuget-inspector/detect-nuget-inspector/Inspection/Util/SolutionDirectoryPackagesPropertyLoader.cs

```diff
@@ -0,0 +1,171 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Xml;
+using Synopsys.Detect.Nuget.Inspector.DependencyResolution.Nuget;
+using Synopsys.Detect.Nuget.Inspector.Model;
+
+namespace Synopsys.Detect.Nuget.Inspector.Inspection.Inspectors
+{
+    public class SolutionDirectoryPackagesPropertyLoader : PackageReferenceLoader
+    {
+
+        private String PropertyPath;
+        private NugetSearchService NugetSearchService;
+
+        public SolutionDirectoryPackagesPropertyLoader(String propertyPath, NugetSearchService nugetSearchService)
+        {
+            PropertyPath = propertyPath;
+            NugetSearchService = nugetSearchService;
+        }
+
+        public HashSet<PackageId> Process()
+        {
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+
+            XmlDocument doc = new XmlDocument();
+            doc.Load(PropertyPath);
+
+            XmlNodeList manageCentralPackageVersion = doc.GetElementsByTagName("ManagePackageVersionsCentrally");
+
+            HashSet<PackageId> packageReferences = new HashSet<PackageId>();
+            
+            if (CheckCentralPackageManagementEnabled(manageCentralPackageVersion))
+            {
+                XmlNodeList importNodes = doc.GetElementsByTagName("Import");
+
+                var propsFilePath = GetPropsFilePath(importNodes);
+
+                if (!String.IsNullOrWhiteSpace(propsFilePath) && !propsFilePath.Equals("..\\Directory.Packages.props"))
+                {
+                    Console.WriteLine("Import of Directory.Packages.props file from a non-standard location.");
+                }
```

Comment:
This is a check to see if there is a Import tag in the props then we check if it is correctly importing the root Directory.Packages.props file otherwise we add a warning for the customers to say that we are trying to import from a non-standard location.